h/api/test/schemas_test.py

```diff
@@ -91,6 +91,44 @@ def test_createannotationschema_ignores_input_user(data, authn_policy):
     assert result == {'user': 'acct:jeanie@example.com'}
 
 
+@pytest.mark.parametrize('data,effective_principals,ok', [
+    # No group supplied
+    ({}, [], True),
+
+    # World group
+    ({'group': '__world__'}, [], False),
+    ({'group': '__world__'}, ['group:__world__'], True),
+
+    # Other group
+    ({'group': 'abcdef'}, [], False),
+    ({'group': 'abcdef'}, ['group:__world__'], False),
+    ({'group': 'abcdef'}, ['group:__world__', 'group:abcdef'], True),
+])
+def test_createannotationschema_rejects_annotations_to_other_groups(data,
```

Comment:
I've been naming these like `test_CreateAnnotationSchema_*` 'cos I don't think that lowercasing it helps readability (even though pylint will complain). (Or put the tests for a class in a test class.)